update-versions.sh

```diff
@@ -18,55 +19,89 @@ fi
 
 set -ex
 
+if ! which rg &> /dev/null; then
+    echo "Command `rg` (ripgrep) was not found. Please install it and try again."
+    exit 1
+fi
+
+if ! cargo set-version --help &> /dev/null; then
+    echo "Cargo extension `cargo-edit` is not installed. Please install it by running: cargo install cargo-edit"
+    exit 1
+fi
+
 HEAD=$(git rev-parse HEAD)
 VERSION=$1
 
-CARGO_TOMLS_TO_BUMP=(
-    ./Cargo.toml
-    ./pgx/Cargo.toml
-    ./pgx-utils/Cargo.toml
-    ./pgx-macros/Cargo.toml
-    ./pgx-tests/Cargo.toml
-    ./cargo-pgx/Cargo.toml
-    ./pgx-pg-sys/Cargo.toml
-)
-
-CARGO_TOMLS_TO_SED=(
-    ./cargo-pgx/src/templates/cargo_toml
-    ./nix/templates/default/Cargo.toml
-    ./pgx/Cargo.toml
-    ./pgx-utils/Cargo.toml
-    ./pgx-macros/Cargo.toml
-    ./pgx-tests/Cargo.toml
-    ./cargo-pgx/Cargo.toml
-    ./pgx-pg-sys/Cargo.toml
-    ./pgx-examples/*/Cargo.toml
-    ./Cargo.toml
-)
-
-DEPENDENCIES_TO_UPDATE=(
-    "pgx"
-    "pgx-tests"
-    "pgx-macros"
-    "pgx-pg-config"
-    "pgx-pgx-sys"
-    "pgx-utils"
-    "cargo-pgx"
-)
-
-SEMVER_REGEX="(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"
-
-for cargo_toml in ${CARGO_TOMLS_TO_SED[@]}; do
-    for dependency in ${DEPENDENCIES_TO_UPDATE[@]}; do
-        rg --passthru -N "(?P<prefix>^${dependency}.*\")(?P<pin>=?)${SEMVER_REGEX}(?P<postfix>\".*$)" -r "\${prefix}=${VERSION}\${postfix}" ${cargo_toml} > ${cargo_toml}.tmp || true
+# Use `cargo set-version` to update all main project files
+function update_main_files() {
+    local version=$1
+
+    EXCLUDE_PACKAGES=()
+
+    # Add additional packages to ignore by using the following syntax:
+    # EXCLUDE_PACKAGES+=('foo' 'bar' 'baz')
+
+    # Ignore all packages in pgx-examples/
+    for file in $(find pgx-examples/ -type f -name "Cargo.toml"); do
+        EXCLUDE_PACKAGES+=($(cat $file | rg --multiline '\[package\](.*\n)(?:[^\[]*\n)*name\s?=\s?"(?P<name>[-_a-z]*)"(.*\n)*' -r "\${name}"))
+    done
+
+    echo "Excluding the following packages:"
+    for p in "${EXCLUDE_PACKAGES[@]}"; do
+        echo " * $p"
+    done
+    echo ""
+
+    cargo set-version $(echo "${EXCLUDE_PACKAGES[@]/#/--exclude }") --workspace $version
```

Comment:
This is trying to substitute an array into a bunch of strings and then feed it through echo to flatten it and then reexpand it by embedding it as a command evaluation and... it would be simpler to allow the shell to do the thing it does.
```suggestion
    # shellcheck disable=2068 # allow the shell to split --exclude from each EXCLUDE_PACKAGES value
    cargo set-version ${EXCLUDE_PACKAGES[@]/#/--exclude } --workspace "$version"
```